web/modules/custom/ingredient_validation/src/Plugin/Validation/Constraint/IngredientConstraintValidator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ingredient_validation\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the Ingredients field on the Product CT.
+ */
+final class IngredientConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint): void {
+    $lines = explode(PHP_EOL, strval($value->value));
```

Comment:
```suggestion
    $lines = explode(PHP_EOL, (string) $value->value);
```
Maybe casting to string would be better.

The field this contraint and validator are attached to is a multi-value field. The ->value property accessor will only return the first value of the multi-value field. You have two choices here:

1. Take into account that you are validating a multi-value field.
2. Change the field to be a single-value field (discuss this option with your team, there might be a valid reason for using a multi-value field here). If the field already has multiple items in it on the production server... well... then option 1 is the less painful (and only practical) path.

Reply:
We've already agreed on that we'll have one field_ingredient and Gábor Törtei will change this in a PR later.